### ISSUE TITLE
InfluxDB 2.7 does not support deleting data by field.

### ISSUE THREAD

**Issue by teremok87** [solectrus/solectrus#3045]
Hallo Geord,

ich habe ein echtes problem. Ich habe an meiner PV Anlage Änderungen vorgenommen. Da man bei solectrus nicht beliebig viel einzel MPPTs  separat in die Datenbank schreiben kann, muss ich die Summe aller erträge in Node - Red zusammenrechnen. Leider hat mir beider Berechnung ein Fehler unterlaufen. Ich wollte die falschen Daten rausnehmen und durch korrekten ersetzen. Leider geht es nicht.

Ich nutze InfluxDB 2.7. Laut der Angabe auf der InfluxDB homepage kann man die Daten nur bei _measurment oder custom_tag löschen.

Du lägst aber die Daten bei _field ab. Da funktioniert das Löschen eben nicht.

Wir dürfen ja seit einer Weile eigenen Buket nutzen. Könntest uns es bitte ermöglichen die Measurments für PV Power, Haus Last, SoC usw. ebenfalls über Enviroment Variable selber zu defnieren. 

Hier wäre ein beispiel meiner Seits:

Buket: solectrus
           Measurment:   inverter_power
           Measurment:   grid_power_minus
           Measurment:   grid_power_plus
           Measurment:   house_power
           Measurment:   bat_fuel_charge
           Measurment:   bat_power_minus
           Measurment:   bat_power_plus
           Measurment:   wallbox_charge_power
           Measurment:   case_temp

Vielen Dank im Voraus!

P.S: Ich freue mich schon auf Tibber integration :)

**Comment by ledermann** [comment]
Ja, grundsätzlich ist es so, dass die DELETE-Funktion von InfluxDB keinen Filter auf einzelne Fields vorsieht. Man kann nur:

- Gesamtes Measurement löschen
- Alle Measurements mit einem bestimmten Tag löschen (SOLECTRUS verwendet übrigens keine Tags)
- Alle Daten eines bestimmten Zeitraums löschen

Das sind Einschränkungen, die aber nur selten von praktischer Relevanz sind. In deinem Fall sieht das anders aus, da du von fehlerhaft geschriebenen Daten berichtest. Je nachdem, wie der Fehler aussieht, kannst du Daten aber überschreiben, was natürlich nur sinnvoll ist, wenn Measurement und Field korrekt sind und nur der Value falsch ist.

Ansonsten besteht die Möglichkeit, die Daten in ein neues Bucket zu kopieren. Dieser Vorgang erlaubt meines Wissens ein Filtern nach Field, ich habe das aber selbst noch nicht ausprobiert. Danach kann das alte Bucket weggeworfen werden und die fehlerhaften Daten sind bereinigt.

Für die Zukunft wird SOLECTRUS genauer konfiguriert werden können, man kann also zukünftig für jeden "Sensor" (= neues Konzept) angeben, in welchem Measurement und welchem Field der Wert zu finden ist:
https://github.com/solectrus/solectrus/pull/2836

Der MQTT-Collecter lässt sich zukünftig analog einstellen:
https://github.com/solectrus/mqtt-collector/pull/137

Der SENEC-Collector hingegen wird weiterhin alles in EIN Measurement schreiben. Hier sehe ich keine Notwendigkeit, das anders zu handhaben - die Daten kommen ja alle aus der gleichen Quelle.

Das Konzept der Sensoren wird nützlich sein, um später auch weitere Daten nach SOLECTRUS pumpen zu können, auch Tibber-Preise u.a.
